lib/chai/core/assertions.js

```diff
@@ -1532,6 +1532,39 @@ module.exports = function (chai, _) {
   });
 
   /**
+   * ### .oneOf(list)
+   *
+   * Assert that a non-array, non-object value appears somewhere in the flat array `list`.
+   *
+   *     expect('a').to.be.oneOf(['a', 'b', 'c']);
+   *     expect(9).to.not.be.oneOf(['z']);
+   *
+   * @name oneOf
+   * @param {Array<*>} list
+   * @param {String} message _optional_
+   * @api public
+   */
+
+  function oneOf (list, msg) {
+    if (msg) flag(this, 'message', msg);
+    var expected = flag(this, 'object');
+    new Assertion(list).to.be.an('array');
+    new Assertion(expected).to.not.be.an('array');
+    new Assertion(expected).to.not.be.an('object');
```

Comment:
I wonder if this requirement really matters? Would it be so wrong to have `oneOf` work with object or arrays? Referential equality would be fine for an initial implementation - we can add `.deep.oneOf()` later if we need to.